### DISCUSSION TITLE
How to visualize strings in the QtCreator debugger variable window

### DISCUSSION THREAD

**Discussion by marip8** [ros-industrial/ros_qtc_plugin#513] [Show and tell]
Often, strings aren't visualized correctly in the Qt debugger variable window by default. This [comment on the bug report](https://bugreports.qt.io/browse/QTCREATORBUG-22753?focusedCommentId=483030&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-483030) provides a solution for visualizing them correctly.

Here is the relevant snippet from the report:

> You can try to fix **"/usr/share/qtcreator/debugger/stdtypes.py"**. As they use "hard-code assumption on member position" it seems like not working everywhere. In my case - Linux x64, gcc 9.1 it works exactly like you described: string not accessible
>
> So find function **def qdumpHelper_std__string(d, value, charType, format):**
>
> And change **(size, alloc, refcount) = d.split("ppp", data - 3 * d.ptrSize())** to **(size, alloc, refcount) = d.split("ppp", value.address() + d.ptrSize())**
>
> Also comment **d.check(0 <= size and size <= alloc and alloc <= 100\*1000\*1000)** ...

Note, the file `qtcreator/debugger/stdtypes.py` is located in the directory where QtCreator was installed, which may or may not be `/usr/share` depending on how you installed QtCreator

**Comment by christian-rauch** [comment]
How is this related to the ROS plugin? Isn't this something that has to be fixed in Qt Creator itself?

**Reply by marip8** [reply]
This post was intended as helpful pointer for an issue I (and many of my coworkers) have run into using QtCreator. This plugin is our entry point into QtCreator, so I thought the discussions page would be a convenient place to put tips like this.

We have similar tips (about enabling the debugger, etc.) in the docs for this repo. Maybe we should ultimately move this tip there as well.

**Reply by christian-rauch** [reply]
Ok. I initially thought you placed this in the wrong repo.

If you are interested in collecting these tips and having them conveniently available via this repo somehow, you could add them to the wiki (https://github.com/ros-industrial/ros_qtc_plugin/wiki).
